crates/cxx-qt/src/lib.rs

```diff
@@ -76,3 +76,120 @@ pub trait Threading: Locking + Sized {
     #[doc(hidden)]
     fn threading_drop(cxx_qt_thread: &mut CxxQtThread<Self>);
 }
+
+/// This trait can be implemented inside the [`#[cxx_qt::bridge]`](bridge) on any [CxxQtType] to define a
+/// custom constructor in C++ for the QObject.
+///
+/// The `Arguments` must be a tuple of CXX types that will be the arguments to the constructor in C++.
+///
+/// Note that implementing this trait will stop Cxx-Qt from generating a default constructor.
+///
+/// # Minimal Example
+/// ```
+/// #[cxx_qt::bridge]
+/// mod ffi {
+///     // Struct without `Default` implementation
+///     #[cxx_qt::qobject]
+///     pub struct MyStruct {
+///         pub integer: i32,
+///         pub string: String
+///     }
+///
+///     impl cxx_qt::Constructor<(i32, String), NewArguments=(i32, String)> for qobject::MyStruct {}
+/// }
+///
+/// impl cxx_qt::Constructor<(i32, String)> for qobject::MyStruct {
+///     type BaseArguments = (); // Will be passed to the base class constructor
+///     type InitializeArguments = (); // Will be passed to the "initialize" function
+///     type NewArguments = (i32, String); // Will be passed to the "new" function
+///
+///     fn route_arguments(args: (i32, String)) -> (
+///         Self::NewArguments,
+///         Self::BaseArguments,
+///         Self::InitializeArguments
+///     ) {
+///         (args, (), ())
+///     }
+///
+///     fn new((integer, string): (i32, String)) -> MyStruct {
+///         MyStruct {
+///             integer,
+///             string
+///         }
+///     }
+/// }
+///
+/// # // Note that we need a fake main function for doc tests to build.
+/// # fn main() {}
+/// ```
+///
+/// # Pseudo Code for generated C++ Constructor
+/// You can imagine this trait as creating a constructor roughly like this:
+/// ``` cpp
```

Comment:
remove the space before cpp ?

Reply:
Done.

Also updated the doc comment to reflect the change to a declarative API.